src/plugins/intel_gna/src/transformations/insert_identity_layer.hpp

```diff
@@ -68,6 +68,73 @@ class IdentityCandidatesCleanup : public ov::pass::ModelPass {
     bool run_on_model(const std::shared_ptr<ov::Model>& m) override;
 };
 
+/**
+ * @brief Inserts identity for precision agnostic (or FQ) concat inputs
+ * Example model:
+ *
+ *                      Parameter
+ *                          |
+ *   Functional      Prec-Agnostic or FQ
+ *              \       /
+ *               Concat
+ *                  |
+ *               Result
+ *
+ * For the above model, during scale factors propagation, when left Concat input
+ * will be selected as a scale factors source, then algorithm will not be able to
+ * apply the scale factors to the right Concat input and will throw exception.
+ * InsertIdentityForPrecAgnosticConcatInput pass adds Identity layer, which
+ * is capable of storing scale factors, so the scale factors propagation can proceed:
+ *
+ *                    Parameter
+ *                        |
+ *                  Prec-Agnostic or FQ
+ *                        |
+ *      Functional     Identity
+ *              \       /
+ *               Concat
+ *                  |
+ *               Result
+ */
+class InsertIdentityForPrecAgnosticConcatInput : public ov::pass::ModelPass {
+    /**
```

Comment:
1. as it was agreed in coding convention private members and methods declaration should be done after public.
2. Are you sure you want to expose this methods in pass declaration. Almost all methods in passes which are not constructors are declared as free functions in source file(static or inline to avoid missing forward declaration warning).

Reply:
1. Agreed / changed
2. I'm sure ;)

Reply:
When I have implemented the needed functionality for InsertIdentityForPrecAgnosticConcatInput pass, the code was far too long to be easily readable and understandable for others. So I divided it into shorter, functional parts. And because still those parts functionally are inseparably related to InsertIdentityForPrecAgnosticConcatInput, I just created in-class methods. I didn't want to create static functions in cpp file, which contains implementation for many classes, because these functions are not universal and are used just by one of these classes for specific purpose.